stix_shifter_modules/nozomi_vantage/stix_transmission/connector.py

```diff
@@ -0,0 +1,272 @@
+
+    def __init__(self, connection, configuration):
+        self.api_client = APIClient(connection, configuration)
+        self.logger = logger.set_logger(__name__)
+        self.connector = __name__.split('.')[1]
+        self.expired_token_count = 0
+
+    async def create_results_connection(self, query, offset, length, metadata=None):
+        """
+        Fetching the results using query, offset and length
+        :param query: str, Data Source query
+        :param offset: str, Offset value
+        :param length: str, Length value
+        :param metadata: dict
+        :return: return_obj, dict
+        """
+        data = []
+        try:
+            result_count, total_records, page_number, page_size = self.handle_metadata_and_page_size(offset, length,
+                                                                                                     metadata)
+            # Generating jwt token
+            jwt_token, return_obj = await self.__get_token()
+            if return_obj:
+                return return_obj
+
+            if (result_count == 0 and page_number == 1) or \
+                    (page_number and result_count < self.api_client.result_limit):
+
+                while result_count < total_records:
+                    query_url = f'{query}&page={page_number}&count={page_size}'
+                    response_wrapper = await self.api_client.get_search_results(query_url, jwt_token)
+                    response_dict, return_obj = self.handle_api_response(response_wrapper)
+
+                    if return_obj:
+                        return return_obj
+
+                    # Generate a new token only the token_generated_count is 1.
+                    # If it is more than one, it has already been generated.
+                    if self.expired_token_count == 1:
+                        jwt_token, return_obj = await self.__get_token()
+                        self.expired_token_count += 1
+                        if return_obj:
+                            return return_obj
+                        continue
+
+                    return_obj['success'] = True
+                    data += response_dict['result']
+                    result_count += len(response_dict['result'])
+
+                    if response_dict['total'] <= result_count or len(data) == 0:
+                        page_number = None
+                        break
+
+                    remaining_records, page_size = self.get_page_size(result_count, total_records, page_size)
+                    if not remaining_records:
+                        break
+
+                    page_number += 1
+
+                return_obj = self.handle_data(data, metadata, offset, total_records, return_obj)
+
+                if return_obj.get('data'):
+                    if page_number and result_count < self.api_client.result_limit:
+                        return_obj['metadata'] = {"result_count": result_count, "page_number": page_number+1,
```

Comment:
Should only return back the remaining_page_data_size and the current page number (would be page_number -1 because it's incremented on the last loop in the while).

Reply:
Currently, metadata contains result_count and page_number.
Incrementing page_number at the end of the loop; when exiting the loop, it will always be on the current page.
because of that incrementing the page_number in the metadata. So on the next call, we can directly use the page number.

Reply:
This comment was made assuming you had static unchanging page_size.

Reply:
page_size is now static (configured value).